File: my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
class smartphonenode:public rclcpp::Node
{
    public:
    smartphonenode():Node("smartphone")
    {
        subscriber_ = this->create_subscription<example_interfaces::msg::String>("robot_news",10,std::bind(&smartphonenode::callbackrobotnews,this,std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(),"Smartphone has been started.");
    }
    private:

        void callbackrobotnews(const example_interfaces::msg::String::SharedPtr msg)
        {
            RCLCPP_INFO(this->get_logger(),"I heard: '%s'",msg->data.c_str());
        }

        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc,char** argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<smartphonenode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class smartphonenode(Node):
    def __init__(self):
        super().__init__('smartphone')
        self.subcriber=self.create_subscription(String, 'robot_news', self.callback_robot_news, 10)
        self.get_logger().info('Smartphone has been started')

    def callback_robot_news(self,msg):
        self.get_logger().info(msg.data)

def main(args=None):
    rclpy.init(args=args)
    node = smartphonenode()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == '__main__':
    main()